File: SDKs/zuora.com/2021-08-20/python/src/sdk/models/shared/saveresult.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import actionserrorresponse as shared_actionserrorresponse


@dataclass_json
@dataclasses.dataclass
class SaveResult:
    errors: Optional[list[shared_actionserrorresponse.ActionsErrorResponse]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Errors') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Id') }})
    success: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Success') }})
    

File: SDKs/zuora.com/2021-08-20/typescript/src/sdk/models/shared/saveresult.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ActionsErrorResponse } from "./actionserrorresponse";



export class SaveResult extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=Errors", elemType: ActionsErrorResponse })
  errors?: ActionsErrorResponse[];

  @SpeakeasyMetadata({ data: "json, name=Id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=Success" })
  success?: boolean;
}
